network/dime_message.ksy

```diff
@@ -0,0 +1,115 @@
+meta:
+  id: dime_message
+  title: DIME (Direct Internet Message Encapsulation) Message
+  file-extension:
+    - dim
+    - dime
+  xref:
+    mime: application/dime
+    wikidata: Q1227457
+  license: CC0-1.0
+  bit-endian: be
+  endian: be
+  encoding: ASCII
+doc: |
+  Direct Internet Message Encapsulation (DIME)
+  is an old Microsoft specification for sending and receiving
+  SOAP messages along with additional attachments,
+  like binary files, XML fragments, and even other
+  SOAP messages, using standard transport protocols like HTTP.
+  
+  Sample file: `curl -L
+  https://github.com/kaitai-io/kaitai_struct_formats/files/5890499/scanner.dump.gz
+  | gunzip -c > scanner.dump`
+doc-ref: 
+  - https://tools.ietf.org/html/draft-nielsen-dime-02
+  - https://docs.microsoft.com/en-us/archive/msdn-magazine/2002/december/sending-files-attachments-and-soap-messages-via-dime
+  - http://imrannazar.com/Parsing-the-DIME-Message-Format
+seq:
+  - id: records
+    type: record
+    repeat: eos
+types:
+  padding:
+    doc: padding to the next 4-byte boundary
+    seq:
+      - id: boundary_padding
+        size: (- _io.pos) % 4
+  option_field:
+    doc: the option field of the record
+    seq:
+      - id: option_elements
+        type: option_element
+        repeat: eos
+  option_element:
+    doc: one element of the option field
+    seq:
+      - id: element_format
+        type: u2
+      - id: len_element
+        type: u2
+      - id: element_data
+        size: len_element
+  record:
+    doc: each individual fragment of the message
+    seq:
+      - id: version
+        doc: DIME format version (always 1)
+        type: b5
```

Comment:
Since 0.9, this can be checked using the `valid` key (kaitai-io/kaitai_struct#435):

```suggestion
      - id: version
        doc: DIME format version
        type: b5
        valid: 1
```

_Alternatively:_ 
```ksy
      - id: version
        doc: DIME format version
        type: b5
        valid:
          eq: 1
```

Reply:
Didn't know this, cool, thanks.

Reply:
Probably it shouldn't. The format has reserved fields, and the point to have reserved fields is forward compatibility.

Reply:
Yeah, that is true. But the format was also withdrawn before it even made RFC in 2002. There most certainly won't be other version (at least official). I don't think Microsoft will dig out this 20 years old spec and suddently make a 2.0. 

But you are right, there *could* be other versions. I don't know what the Kaitai's approach is on things like that.

Reply:
> But you are right, there _could_ be other versions. I don't know what the Kaitai's approach is on things like that.

I'm not sure where I stand. For best compatibility, it probably makes sense to follow the [robustness principle](https://en.wikipedia.org/wiki/Robustness_principle) (_"Be conservative in what you do, be liberal in what you accept from others"_), which would mean to accept _any_ version specifier, but generate only `1` (after https://github.com/kaitai-io/kaitai_struct/issues/27 is implemented).

So the existing approach is totally OK I'd say.